### DISCUSSION TITLE
Curiously wondering how @entangle() works...

### DISCUSSION THREAD

**Discussion by yob-yob** [livewire/livewire#3683] [1. Help] [answered]
Hi.. I'm trying to figure out how livewire pulled `entanglement`

so here's my progress

`@entangle` is a blade directive that generates this code
```js
window.Livewire.find('6BRbHAi1kJgVwVD11kZG').entangle("record.main_content")
```
`6BRbHAi1kJgVwVD11kZG` => from what I know is kind-of hash that is used to locate the livewire component....
`record.main_content` => is a property on that component (you got me, I'm using filamentadmin)

so using the code above.... I tried to log the output on the console...

```js
window.Livewire.find('6BRbHAi1kJgVwVD11kZG').entangle("record.main_content")
  {isDeferred: true, livewireEntangle: "record.main_content"}
    defer: (...)
    isDeferred: false
    livewireEntangle: "record.main_content"
    get defer: ƒ defer()
    [[Prototype]]: Object
```

opening `defer: (...)` will only open a recursive output of 
```
window.Livewire.find('6BRbHAi1kJgVwVD11kZG').entangle("record.main_content").defer
```

# Now the questions: 🤔
1. How does it works ?
2. Why is it returning a recursive output ? 
3. How can I get the REAL - Value using the console ?

**Comment by joshhanley** [comment, marked as answer]
@yob-yob the object that you are getting back from `.entangle` is just a placeholder object with some data.

That object gets saved into the Alpine property that you are entangling to.

Then when Livewire boots, it will run through the Alpine properties and look for any that have that placeholder object, it then configures both Alpine and Livewire watchers, and then get's the value of the Livewire property and saves it to the Alpine property.

You can find the code that does that here 

https://github.com/livewire/livewire/blob/8cac40cf967bd7d64f3df349e014c4f4bd1e0445/js/component/SupportAlpine.js#L76-L161

That is why your console log won't return any actual value data in that object, because there is none in that placeholder object.

To get the real value just use the `$wire` Alpine helper and get your property name like `$wire.someProperty`.

The question is though, what are you trying to achieve?

Hope this helps!

**Comment by yob-yob** [comment]
thank you sooo much for explaining and giving me resources... 

there's no goal actually, I was just amazed how `alpine` and `livewire` syncs data using `entanglement` that feature is so dope...

I still do have questions.. but because you've provided me a resource.. I think I can probably trace the answer from their..

**Reply by joshhanley** [reply]
No worries! Have a play and if you have any more questions, feel free to ask, I'd be happy to answer them if I can 🙂
